cmd/manager/main.go

```diff
@@ -123,6 +123,17 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "ClusterExtension")
 		os.Exit(1)
 	}
+	if features.OperatorControllerFeatureGate.Enabled(features.EnableExtensionApi) {
+		if err = (&controllers.ExtensionReconciler{
+			Client:         cl,
+			BundleProvider: catalogClient,
+			Scheme:         mgr.GetScheme(),
+			Resolver:       resolver,
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create controller", "controller", "Extension")
+			os.Exit(1)
+		}
+	}
```

Comment:
One of the implications with not even adding a reconciler for the `Extension` API to the controller manager is that if this feature gate is disabled, creating an `Extension` resource will silently do nothing. From a UX perspective I think it would make sense to still add the reconciler but when the feature gate is disabled we simply add a status condition that states something along the lines of "Reconciliation skipped. feature gate for the Extension API is disabled. The Extension API is an alpha feature that is still in development. To enable reconciliation use the flag --feature-gates=EnableExtensionApi=true".

Doing this would at least signal to an end user that it is currently in a "do nothing" state and how to enable it.

Reply:
That being said, I'm also not sure of any API versioning constraints that come with adding a status condition for this. Definitely something to think about regarding the pros/cons

Reply:
Yeah, I'm a bit torn. on where to put the feature gate.

Reply:
The ideal scenario would be to not have the Extension CRD applied if the feature gate is disabled - but I don't think that's something which is possible here, since the CRDs are applied before the controller pod is created. Instead of setting up the reconciler by default and adding the overhead of watching Extension resources, what do you all think about adding a log/warning during startup, mentioning the value of the feature gate flag, and explicitly stating that - "feature gate flag for `ExtensionAPI` set to false, no action will be taken by operator controller if Extension object is created". That should to some extent satisfy the UI difficulties without having us to take more effort on it for now?

Reply:
@varshaprasad96 I think that assumes a user who might create the Extension resource would have an understanding to go and look at the logs of the operator-controller pod and explicitly find that singular log message.

Throwing a message on the status is more immediately visible IMO and to me doesn't feel like a bunch of overhead. I'm mainly approaching this from what I would want as a user and I would much prefer seeing a status be set than having to find the pod that is running the controller manager, get the logs, and search the logs for any indication as to why nothing happened

Reply:
@everettraven has a point, if your extension isn't running, you're going to look at your extension first, and not necessarily a pod to which you don't have log access to.

Reply:
Some thoughts on why this suggestion:

> I think that assumes a user who might create the Extension resource would have an understanding to go and look at the logs of the operator-controller pod and explicitly find that singular log message.

I would expect that users who create Extension resources have the understanding of troubleshooting at this stage, which includes checking logs when something doesn't work as expected - this is particularly considering the fact that afaik there is no one trying to use v1 in production in near future, at least till we make kapp-ctrl switch -scheduled soon. 

That being said, I understand and completely agree that having just a log line - that too in an o-c pod that the user trying to install the package, ideally should not concern with - is not the perfect UX. This is to avoid - both the overhead of having an additional reconciler watch extension and other relevant resources on cluster which is not necessary, and us spending time on an API field that could turn out to be temporary in very near future. On a quick search upstream found an [example](https://github.com/Kong/kubernetes-ingress-controller/blob/d5cb0fa57d7d6f9360f1b8f58859c6b633cda1b6/internal/manager/controllerdef.go#L278-L279) of one of the optional controllers enabled only when the feature gate is present.